### ISSUE TITLE
[BUG][java][Python] Integration tests failing on main

### ISSUE THREAD

**Issue by mugdhapattnaik** [GoogleCloudPlatform/dataproc-templates#577]
The default value of Spark runtime version that is being used is 2.0 and the default Scala version for this Spark version is 2.13. However the templates cannot run with Scala 2.13 because the connectors were meant to run with Scala 2.12

**Comment by hhasija** [comment]
@mugdhapattnaik Please do start working on this with high priority.

**Comment by TheTravellingSalesman** [comment]
Is this tied to issues that I'd be seeing with the spark-bigquery-connector failing with 
```
py4j.protocol.Py4JJavaError: An error occurred while calling o74.load.
: java.util.ServiceConfigurationError: org.apache.spark.sql.sources.DataSourceRegister: com.google.cloud.spark.bigquery.BigQueryRelationProvider Unable to get public no-arg constructor
```
?

I figure that that jar package isn't upgraded to work with Java 17 yet, but even when using the 1.1.1 runtime config on Java 11, the behavior of that jar package has changed, and my guess is because the 1.1.1 runtime has upgraded it to 0.28.0, and overwrites any older version you try to set through the `jar_file_uris` property on the `DataprocPySparkBatchOp`.

If this is the wrong place for this, apologies!

**Comment by hhasija** [comment]
Yes @TheTravellingSalesman , this is in resolution to that issue only, which you are facing with Big Query connector. More over, this is the right place for discussing that as well.

We are on top of that issue, and hot fix would be up at max by Tuesday EOD.

**Comment by hhasija** [comment]
@TheTravellingSalesman The templates are now working fine. Can you please check once form your side as well ?

**Comment by TheTravellingSalesman** [comment]
Hey there, sorry for the holdup. Yes, my serverless dataproc components are working just fine now, but I've only tested using version `1.0.29` since the fix was merged. 

I'll give the other compatible versions a try (1.1.x) and report back if anything isn't behaving as expected.

**Comment by surjits254** [comment]
Hi,

The fix is currently 1.0.29. We are working on upgrading this to 1.1.

Thanks
